docs/useRegisterMutationMiddleware.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: default
+title: "useRegisterMutationMiddleware"
+---
+
+# `useRegisterMutationMiddleware`
+
+This hook allows to register a mutation middleware function.
+
+Middleware functions allow you to hook into a mutation process to perform various actions before or after the main mutation action (create or update). They allow similar use cases as the [`withLifeCycleCallbacks`](./withLifecycleCallbacks.md) except they are declared at the component level.
+
+**Tip** This hook will unregister the middleware function in its cleanup phase (for instance when the component that called it is unmounted).
+
+## Usage
```

Comment:
You should explain the notion of middleware and `next` function before. This example isn't self-explanatory.

Reply:
I added a paragraph about them below